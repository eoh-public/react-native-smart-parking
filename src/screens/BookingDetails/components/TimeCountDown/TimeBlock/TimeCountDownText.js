import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';

import Text from '../../../../../commons/Text';
import { Colors } from '../../../../../configs';
import { TESTID } from '../../../../../configs/Constants';

const TimeCountDownText = memo(({ number, color }) => {
  return (
    <View style={styles.container}>
      <View style={styles.topColor} />
      <View style={styles.bottomColor} />
      <Text
        testID={TESTID.TIME_COUNT_DOWN_TEXT}
        type="H2"
        style={styles.textNumber}
        color={color}
        semibold
      >
        {number}
      </Text>
    </View>
  );
});

export default TimeCountDownText;

const styles = StyleSheet.create({
  container: {
    width: 32,
    height: 48,
    borderRadius: 2,
    marginRight: 4,
    shadowColor: Colors.Black,
    shadowOffset: {
      width: 2,
      height: 3,
    },
    shadowOpacity: 0.1,
    shadowRadius: 2,

    elevation: 1,
    alignItems: 'center',
  },
  topColor: {
    position: 'absolute',
    width: 32,
    height: 24,
    backgroundColor: Colors.Gray3,
    borderTopLeftRadius: 2,
    borderTopRightRadius: 2,
  },
  bottomColor: {
    width: 32,
    height: 24,
    position: 'absolute',
    top: 24,
    backgroundColor: Colors.Gray4,
    borderBottomLeftRadius: 2,
    borderBottomRightRadius: 2,
  },
  textNumber: {
    marginTop: 8,
    textAlign: 'center',
    width: 32,
  },
});
