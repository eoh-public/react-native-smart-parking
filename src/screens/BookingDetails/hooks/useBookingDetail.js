import { useCallback, useEffect, useState } from 'react';
import { API } from '../../../configs';
import { axiosGet } from '../../../utils/Apis/axios';
import { transformDatetime } from '../../../utils/Converter/time';

const useBookingDetail = (id, fetchNow) => {
  const [showThanks, setShowThanks] = useState(false);
  const [loading, setloading] = useState(false);
  const [bookingDetail, setBookingDetail] = useState({
    extend_at: [],
  });

  const getBookingDetail = useCallback(async () => {
    setloading(true);
    const { data, success } = await axiosGet(API.BOOKING.DETAIL(id));
    if (success) {
      transformDatetime(data, [
        'arrive_at',
        'leave_at',
        'book_at',
        'pay_before',
        'pay_at',
        'extend_at',
      ]);
      setBookingDetail(data);
    }
    setloading(false);
  }, [id]);

  useEffect(() => {
    if (fetchNow) {
      getBookingDetail();
    }
  }, [fetchNow, getBookingDetail]);

  useEffect(() => {
    if (!bookingDetail.start_countdown || !bookingDetail.time_remaining) {
      return;
    }

    const timeout = bookingDetail.time_remaining * 1000; // seconds to milliseconds

    const timeHandler = setTimeout(() => {
      getBookingDetail();
    }, timeout);

    return () => clearTimeout(timeHandler);
  }, [bookingDetail, getBookingDetail]);

  useEffect(() => {
    if (!bookingDetail.start_countdown || bookingDetail.is_paid) {
      return;
    }
    const total_time = bookingDetail.num_of_hour_parking * 3600;
    const time_left = bookingDetail.time_remaining;
    let timeHandler;
    if (time_left > total_time - 900) {
      const timeout = time_left - (total_time - 900) + 5;
      timeHandler = setTimeout(() => {
        getBookingDetail();
      }, timeout * 1000);
    }
    return () => clearTimeout(timeHandler);
  }, [bookingDetail, getBookingDetail]);

  const onRefresh = useCallback(() => {
    getBookingDetail();
  }, [getBookingDetail]);

  const onCloseThanks = useCallback(() => {
    setShowThanks(false);
  }, []);

  const onShowThanks = useCallback(() => {
    setShowThanks(true);
    getBookingDetail();
  }, [getBookingDetail]);

  return {
    loading,
    onRefresh,
    bookingDetail,
    getBookingDetail,
    showThanks,
    onCloseThanks,
    onShowThanks,
  };
};

export default useBookingDetail;
