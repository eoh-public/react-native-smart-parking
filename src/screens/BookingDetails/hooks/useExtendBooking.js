import { useCallback, useState } from 'react';
import { API } from '../../../configs';
import { axiosGet, axiosPost } from '../../../utils/Apis/axios';
import moment from 'moment';

const useExtendBooking = (id) => {
  const [showExtend, setShowExtend] = useState(false);
  const [showChecking, setShowChecking] = useState(false);
  const [hour, setHour] = useState(1);
  const [extendInfo, setExtendInfo] = useState({
    last_leave_at: moment(),
  });

  const createExtendBooking = useCallback(
    async (onCreateBookingSuccess) => {
      const body = { hour_extend: hour };
      const res = await axiosPost(API.BOOKING.EXTEND(id), body);
      onCreateBookingSuccess(res);
      setHour(1);
      setShowExtend(false);
    },
    [hour, id]
  );

  const onHideChecking = useCallback(() => {
    setShowChecking(false);
  }, []);

  const onCancelExtend = useCallback(() => {
    setShowExtend(false);
  }, []);

  const onShowExtend = useCallback(async () => {
    setShowChecking(true);
    const { success, data } = await axiosGet(API.BOOKING.EXTEND_INFO(id));
    if (success) {
      data.last_leave_at = moment(data.last_leave_at);
      setExtendInfo(data);
      setTimeout(() => {
        setShowExtend(true);
      }, 1000); // issue https://github.com/react-native-modal/react-native-modal/issues/30
    }
    setShowChecking(false);
  }, [id]);

  return {
    showExtend,
    showChecking,
    extendInfo,
    onHideChecking,
    onCancelExtend,
    onShowExtend,
    hour,
    setHour,
    setShowExtend,
    createExtendBooking,
  };
};

export default useExtendBooking;
