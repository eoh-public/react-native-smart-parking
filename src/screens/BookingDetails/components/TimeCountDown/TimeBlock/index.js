import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';

import { Colors } from '../../../../../configs';
import Text from '../../../../../commons/Text';
import TimeCountDownText from './TimeCountDownText';

const TimeBlock = memo(({ title, time, color }) => {
  const convert = time.length > 1 ? time : '0' + time;
  return (
    <View style={styles.container}>
      <View style={styles.time}>
        <TimeCountDownText number={convert.substring(0, 1)} color={color} />
        <TimeCountDownText number={convert.substring(1, 2)} color={color} />
        {time.length >= 3 && (
          <TimeCountDownText number={convert.substring(2)} color={color} />
        )}
      </View>
      <Text type="Body" color={Colors.Gray8}>
        {title}
      </Text>
    </View>
  );
});

export default TimeBlock;
const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  time: {
    flexDirection: 'row',
  },
});
