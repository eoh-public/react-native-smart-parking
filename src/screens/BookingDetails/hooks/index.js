import useBookingDetail from './useBookingDetail';
import useExtendBooking from './useExtendBooking';

export { useBookingDetail, useExtendBooking };
