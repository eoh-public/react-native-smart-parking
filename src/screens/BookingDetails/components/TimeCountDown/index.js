import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { t } from 'i18n-js';

import { Colors, AppRNConfig } from '../../../../configs';
import Text from '../../../../commons/Text';
import TimeBlock from './TimeBlock';
import { useCountDown } from '../../../../hooks/SmartParking';

const TimeCountDown = memo(
  ({ time_remaining, start_countdown, is_violated, getBookingDetail }) => {
    const { countDown, timeLeft } = useCountDown(
      time_remaining,
      false,
      start_countdown,
      getBookingDetail,
      !is_violated
    );
    const { hours, minutes, seconds } = countDown;

    let color = Colors.Black;
    if (timeLeft < AppRNConfig.MAX_SECONDS) {
      color = Colors.Red6;
    }
    if (is_violated) {
      // revert color
      color = color === Colors.Red6 ? Colors.Black : Colors.Red6;
    }

    return (
      <View style={styles.container}>
        <TimeBlock title={t('hour')} time={hours.toString()} color={color} />
        <Text type="H2" color={Colors.Black} style={styles.text} semibold>
          :
        </Text>
        <TimeBlock
          title={t('minute')}
          time={minutes.toString()}
          color={color}
        />
        <Text type="H2" color={Colors.Black} style={styles.text} semibold>
          :
        </Text>
        <TimeBlock
          title={t('second')}
          time={seconds.toString()}
          color={color}
        />
      </View>
    );
  }
);

export default TimeCountDown;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: Colors.Gray2,
    flex: 1,
    marginTop: 10,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    paddingVertical: 24,
    justifyContent: 'center',
  },
  text: {
    marginLeft: 4,
    marginRight: 8,
    marginTop: 8,
  },
});
