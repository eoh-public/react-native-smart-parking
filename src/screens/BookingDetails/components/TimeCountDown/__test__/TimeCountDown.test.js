import React from 'react';
import { Text } from 'react-native';
import { act, create } from 'react-test-renderer';

import TimeCountDown from '../index';
import { TESTID } from '../../../../../configs/Constants';
import { Colors } from '../../../../../configs';
import { useCountDown } from '../../../../../hooks/SmartParking';
import { t } from 'i18n-js';
import TimeBlock from '../TimeBlock';

jest.mock('../../../../../hooks/SmartParking', () => {
  return {
    useCountDown: jest.fn(),
  };
});

jest.mock('react', () => {
  return {
    ...jest.requireActual('react'),
    memo: (x) => x,
  };
});

describe('Test TimeCountDown', () => {
  let tree;

  test('render TimeCountDown', async () => {
    const countDown = {
      hours: 1,
      minutes: 49,
      seconds: 22,
    };
    const timeLeft = 2962;
    useCountDown.mockImplementation(() => {
      return { countDown, timeLeft };
    });

    act(() => {
      tree = create(<TimeCountDown />);
    });
    const instance = tree.root;
    const texts = instance.findAll(
      (el) =>
        el.props.testID === TESTID.TIME_COUNT_DOWN_TEXT && el.type === Text
    );

    expect(texts[0].props.style[0].color).toEqual(Colors.Black);
    expect(texts[2].props.style[0].color).toEqual(Colors.Black);
    expect(texts[0].props.children).toEqual('0');
    expect(texts[1].props.children).toEqual('1');
    expect(texts[2].props.children).toEqual('4');
    expect(texts[3].props.children).toEqual('9');
    expect(texts[4].props.children).toEqual('2');
    expect(texts[5].props.children).toEqual('2');
  });

  test('render TimeCountDown timeLeft < 15', async () => {
    const countDown = {
      hours: 0,
      minutes: 2,
      seconds: 22,
    };
    const timeLeft = 142;
    useCountDown.mockImplementation(() => {
      return { countDown, timeLeft };
    });

    act(() => {
      tree = create(<TimeCountDown />);
    });
    const instance = tree.root;
    const texts = instance.findAll(
      (el) =>
        el.props.testID === TESTID.TIME_COUNT_DOWN_TEXT && el.type === Text
    );

    expect(texts[0].props.style[0].color).toEqual(Colors.Red6);
    expect(texts[2].props.style[0].color).toEqual(Colors.Red6);
    expect(texts[0].props.children).toEqual('0');
    expect(texts[1].props.children).toEqual('0');
    expect(texts[2].props.children).toEqual('0');
    expect(texts[3].props.children).toEqual('2');
    expect(texts[4].props.children).toEqual('2');
    expect(texts[5].props.children).toEqual('2');
  });

  test('render TimeCountDown timeLeft > 15', async () => {
    const countDown = {
      hours: 0,
      minutes: 2,
      seconds: 22,
    };
    const timeLeft = 910;
    useCountDown.mockImplementation(() => {
      return { countDown, timeLeft };
    });

    act(() => {
      tree = create(<TimeCountDown is_violated />);
    });
    const instance = tree.root;
    const texts = instance.findAllByType(Text);
    expect(texts[0].props.children).toEqual(t('total_violating_time'));
    const timeBlocks = instance.findAllByType(TimeBlock);
    expect(timeBlocks[0].props.color).toEqual(Colors.Red6);
  });
});
