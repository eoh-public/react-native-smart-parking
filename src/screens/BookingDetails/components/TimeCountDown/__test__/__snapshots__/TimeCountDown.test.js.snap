// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TimeCountDown render TimeCountDown 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "borderColor": "#E8E8E8",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 10,
      "paddingVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          1
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.hour" translation]
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 4,
          "marginRight": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    :
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          4
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          9
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.minute" translation]
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 4,
          "marginRight": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    :
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          2
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          2
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.second" translation]
    </Text>
  </View>
</View>
`;

exports[`Test TimeCountDown render TimeCountDown timeLeft < 15 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "borderColor": "#E8E8E8",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 10,
      "paddingVertical": 24,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          0
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.hour" translation]
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 4,
          "marginRight": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    :
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          2
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.minute" translation]
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 4,
          "marginRight": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    :
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          2
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "elevation": 1,
            "height": 48,
            "marginRight": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
            "width": 32,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#E8E8E8",
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "height": 24,
              "position": "absolute",
              "top": 24,
              "width": 32,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#F5222D",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
                "textAlign": "center",
                "width": 32,
              },
            ]
          }
          testID="TIME_COUNT_DOWN_TEXT"
        >
          2
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      [missing "en.second" translation]
    </Text>
  </View>
</View>
`;
